### DISCUSSION TITLE
I'm curious why MILVUS doesn't guarantee the uniqueness of PKs.

### DISCUSSION THREAD

**Discussion by wookiist** [milvus-io/milvus#30809] [Q&A and General discussion] [answered]
I saw a Q&A that said something like this. (https://milvus.io/docs/product_faq.md#Can-vectors-with-duplicate-primary-keys-be-inserted-into-Milvus)

**Can vectors with duplicate primary keys be inserted into Milvus?**
Yes. Milvus does not check if vector primary keys are duplicates.

I'm wondering why milvus cannot support PK uniqueness.

**Comment by xiaofan-luan** [comment, marked as answer]
Good question.
We'd like to guarantee PK uniqueness but for milvus write after read is expensive for now so we decided not to read and check if pk exist.

We are working on a improvement but this need complex architecture change.
